File: CS50x/week2/pset2/readability_DONE/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

float numberOfLetters(string text);
float numberOfWords(string text);
float numberOfSentences(string text);

int totLen;

int main(void)
{
    string text = get_string("Text: ");
    totLen = strlen(text);
    float letters = numberOfLetters(text);
    float words = numberOfWords(text);
    float sent = numberOfSentences(text);
    float L = (letters / words) * 100;
    // printf("Letters per 100 words: %f\n", L);
    float S = (sent / words) * 100;
    // printf("Sentences per 100 words: %f\n", S);
    float level = roundf(0.0588 * L - 0.296 * S - 15.8);
    int ans = level;
    if (ans > 16)
    {
        printf("Grade 16+\n");
    }
    else if (ans < 1)
    {
        printf ("Before Grade 1\n");
    }
    else
    {
        printf("Grade Level: %i\n", ans);
    }
    return 0;
}

// DONE
float numberOfLetters(string text)
{
    float num = 0;
    char letter;
    for (int i = 0; i < totLen; i++)
    {
        letter = text[i];
        // printf("Letter: %c\n", letter);
        if (toupper(letter) != tolower(letter))
        {
            num += 1;
        }
    }
    // printf("Number of Letters: %f\n", num);
    return num;
}

// DONE
float numberOfWords(string text)
{
    //Starting from 1 to account for the fact that there won't be a space after the last word
    float num = 1;
    char letter;
    for (int i = 0; i < totLen; i++)
    {
        letter = text[i];
        if (letter == ' ')
        {
            num += 1;
        }
    }
    // printf("Number of words is: %f\n", num);
    return num;
}

// DONE
float numberOfSentences(string text)
{
    float num = 0;
    char letter;
    for (int i = 0; i < totLen; i++)
    {
        letter = text[i];
        if (letter == '.' || letter == '?' || letter == '!')
        {
            num += 1;
        }
    }
    // printf("Number of sentences is: %f\n", num);
    return num;
}

File: CS50x/week6/pset6/sentimental-readability_DONE/readability.py
def main():
    text = input("Text: ")

    letters = numb_helper(text, numbLetters)
    words = numb_helper(text, numbWords) + 1
    sentences = numb_helper(text, numbSentences)

    print(f"{letters} Letters\n {words} words\n {sentences} sentences")

    L = (letters / words) * 100
    S = (sentences / words) * 100

    level = round(0.0588 * L - 0.296 * S - 15.8)

    if level > 16:
        print("Grade 16+")
    elif level < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {level}")


def numbLetters(char):
    return char.isalpha()

def numbWords(char):
    return char == " "

def numbSentences(char):
    return char == "." or char == "?" or char == "!"

def numb_helper(str, func):
    num = 0
    for i in range(len(str)):
        if func(str[i]):
            num += 1
    return num


main()